File: python examples/Arcade Drive/commands/arcadeDrive.py
from wpilib.command import Command

class ArcadeDrive (Command):

  def __init__ (self, robot):
    self.robot = robot
    self.requires(robot.chassis)
  
  def initialize ():
    pass
  
  def execute (self):
    x = self.robot.dashboard.firstDriver.getX()
    y = self.robot.dashboard.firstDriver.getY()

    self.robot.chassis.setLeftRaw(y - x)
    self.robot.chassis.setRightRaw(y + x)
  
  def isFinished ():
    return False

File: c++ examples/Arcade Drive/cpp/impl/commands/ArcadeDrive.cc
#include "WPILib.h"
#include "../../include/commands/ArcadeDrive.hh"

ArcadeDrive::ArcadeDrive (Robot* robot) {
  this->robot = robot;
  Requires(robot->chassis);
}

bool ArcadeDrive::IsFinished () {
  return false;
}

void ArcadeDrive::Execute () {
  double x = this->robot->dashboard->GetFirstX();
  double y = this->robot->dashboard->GetFirstY();

  this->robot->chassis->SetLeftRaw(y - x);
  this->robot->chassis->SetRightRaw(y + x);
}

void ArcadeDrive::Interupted () {
  this->robot->chassis->SetLeftRaw(0.0);
  this->robot->chassis->SetRightRaw(0.0);
}